encoderpy/onehot_encoder.py

```diff
@@ -1,35 +1,113 @@
 def onehot_encoder(X_train, X_test, cat_columns):
-  """This function encodes categorical variables using the popular onehot method for each category.
-  
-  
-  Parameters
-  ----------
-  X_train : pd.DataFrame
+    """
+    This function encodes categorical variables using the popular onehot method for each category.
+    That is it will convert a category column in the dataframe to dummy binary values like this:
+
+                  X_train = [ index,   num_column ,  cat_column 
+                                 0 ,       1      ,     A      
+                                 1 ,       4      ,     B       ]
+
+        X_train_processed = [ index,   num_column ,  cat_column_A ,  cat_column_B
+                                 0 ,       1      ,        1      ,       0
+                                 1 ,       4      ,        0      ,       1       ]
+
+
+
+    Parameters
+    ----------
+    X_train : pd.DataFrame
           A pandas dataframe representing the training data set containing some categorical features/columns.
-  X_test : pd.DataFrame
+    X_test : pd.DataFrame
           A pandas dataframe representing the test set, containing some set of categorical features/columns.
-  cat_columns : list
-          The names of the categorical features to encode.
-  
-  Returns
-  -------
-  train_processed : pd.DataFrame
+    cat_columns : list
+          The names of the categorical features required to encode.
+
+    Returns
+    -------
+    train_processed : pd.DataFrame
         The training set, with the categorical columns specified by the argument cat_columns
         replaced by their encodings.
-  test_processed : pd.DataFrame
+    test_processed : pd.DataFrame
         The test set, with the categorical columns specified by the argument cat_columns
         replaced by the learned encodings from the training set.
-        
-  Examples
-  -------
-  >>> encodings = onehot_encoder(
-  my_train, 
-  my_test, 
-  cat_columns = ['foo'])
-  
-  >>> train_new = encodings[0]
 
-  """
+    Examples
+    -------
+    >>> encodings = onehot_encoder(
+    my_train, 
+    my_test, 
+    cat_columns = ['foo'])
+
+    >>> train_new = encodings[0]
+    >>> test_new = encodings[1]
+
+    """
   
-  return [train_processed, test_processed]
+    # Check that input is valid
+    if isinstance(X_train, pd.DataFrame) == False:
+        raise Exception("X_train must be a pandas Dataframe type")
+    if isinstance(X_test, pd.DataFrame) == False:
+        raise Exception("X_test must be a pandas Dataframe type")
+    if isinstance(cat_columns, list) == False:
+            raise Exception("cat_columns must be a list type")
+
+
```

Comment:
Make a copy of X_train and X_test.

```suggestion
train_processed = X_train
test_processed = X_test
```